### PR TITLE
[ValueTracking] Handle `not` in `isImpliedCondition`

### PR DESCRIPTION
**Pull request by dtcxzyw**
This patch handles `not` in `isImpliedCondition` to enable more fold in some multi-use cases.

### CODE REVIEW

File: llvm/lib/Analysis/ValueTracking.cpp [hunk 1]
```diff
@@ -8663,21 +8678,21 @@ std::optional<bool> llvm::isImpliedCondition(const Value *LHS, const Value *RHS,
     if (std::optional<bool> Imp =
             isImpliedCondition(LHS, RHS1, DL, LHSIsTrue, Depth + 1))
       if (*Imp == true)
-        return true;
+        return !InvertRHS;
```
Comment (dtcxzyw) [inline llvm/lib/Analysis/ValueTracking.cpp:8681]: Oh, it looks incorrect :(

Comment (goldsteinn) [reply llvm/lib/Analysis/ValueTracking.cpp:8681]: Its confusing enough I think it deserves a comment.

**Comment by goldsteinn** [comment]
LGTM.

File: llvm/lib/Analysis/ValueTracking.cpp (hunk 1 above)
Comment (goldsteinn) [reply llvm/lib/Analysis/ValueTracking.cpp:8681]: NB: I don't any of your tests actually go through this control flow

Comment (goldsteinn) [reply llvm/lib/Analysis/ValueTracking.cpp:8681]: Just a heads up, im working on a follow up patch to this to handle more of the `and/or` cases + more relationships in general (LMK if you are so we can avoid duplicating work).

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**